### ISSUE TITLE
Low numerical accuracy? (new issue)

### ISSUE THREAD

**Issue by nosound2** [pytorch/xla#1085]
## 🐛 Bug

After updating nightly wheel two days ago a new bug appeared, which was not present before. Looks like a numerical issue, a variable that must be less or equal than 1 sometimes can be bigger. In the code below one can see that theoretically cosine variable must be less or equal than one, but there is a case that it is bigger, which causes a crash afterwards.

I am solving it with `torch.clamp(cosine,-1,1)`, reporting it here in case it is interesting to you guys.

```
class ArcMarginProduct(nn.Module):
    def __init__(self, in_feature, out_feature):
        super(ArcMarginProduct, self).__init__()
        self.in_feature = in_feature
        self.out_feature = out_feature
        self.weight = nn.Parameter(torch.Tensor(out_feature, in_feature))
        nn.init.xavier_uniform_(self.weight)

    def forward(self, x):
        output = F.linear(F.normalize(x), F.normalize(self.weight))
        return output
...
self.arc_face_product = ArcMarginProduct(512, num_classes)
...
cosine = self.arc_face_product(x)
...
ipdb> cosine.max()
tensor(1.0002, device='xla:1', grad_fn=<MaxBackward1>)

```

**Comment by dlibenzi** [comment]
Thanks!
I filed an internal bug to track this as we should not be handing out values which are out of the cos's codomain.

**Comment by dlibenzi** [comment]
Hi!
We have not been able to repro this internally.
Any chance you have a small repro so that we can capture the XLA HLO and feed to the team?

**Comment by mruberry** [comment]
Closing since no reproduction. Will keep an eye out and reopen if the issue appears. @nosound2, let us know if you can reproduce the issue with more recent PyTorch/XLA versions.
